### PR TITLE
apply_montage 

### PR DESCRIPTION
**Pull request by aestrivex**
Adds a field to the apply_montage() function to allow imports of montages of different types, which might also need to implement small changes in metadata.

Also add functionality to add localized iEEG electrodes from montage files, in the surface space and using the SEEG coordinate frame. This seems to be the best way to import the types of data that I am working with into .fif format.

### CODE REVIEW

File: mne/channels/layout.py [hunk 1]
```diff
@@ -898,10 +898,10 @@ def read_montage(kind, ch_names=None, path=None, scale=True):
     return Montage(pos=pos, ch_names=ch_names_, kind=kind, selection=selection)
 
 
-def apply_montage(info, montage):
-    """Apply montage to EEG data.
+def apply_montage(info, montage, electrode_type='eeg'):
```
Comment (agramfort) [inline mne/channels/layout.py]: we you the naming ch_type everywhere else. Please rename to ch_type unless someone objects.

File: mne/channels/layout.py [hunk 2]
```diff
@@ -912,8 +912,14 @@ def apply_montage(info, montage):
         The measurement info to update.
     montage : instance of Montage
         The montage to apply.
+    electrode_type : string
+        The type of electrode contained in this montage. Can be eeg or ieeg.
```
Comment (agramfort) [inline mne/channels/layout.py]: Can be "eeg" or "ieeg".

File: mne/channels/layout.py [hunk 3]
```diff
@@ -912,8 +912,14 @@ def apply_montage(info, montage):
...
     """
-    if not _contains_ch_type(info, 'eeg'):
+    supported_types = ('eeg', 'ieeg')
+    if electrode_type not in supported_types:
+        raise ValueError('Montage type not supported')
+
+    if not _contains_ch_type(info, 'eeg') and type == 'eeg':
```
Comment (agramfort) [inline mne/channels/layout.py]: please don't overwrite builtins such as "type"

**Comment by agramfort** [comment]
you'll need to add a test to avoid reducing the coverage of the package

File: mne/channels/layout.py [hunk 4]
```diff
@@ -898,10 +898,10 @@ def read_montage(kind, ch_names=None, path=None, scale=True):
     return Montage(pos=pos, ch_names=ch_names_, kind=kind, selection=selection)
 
 
-def apply_montage(info, montage):
-    """Apply montage to EEG data.
+def apply_montage(info, montage, ch_type='eeg'):
+    """Apply montage.
 
-    This function will replace the EEG channel names and locations with
+    This function will replace the channel names and locations with
```
Comment (agramfort) [inline mne/channels/layout.py]: the EEG or iEEG channel names

File: mne/channels/layout.py [hunk 5]
```diff
@@ -912,8 +912,14 @@ def apply_montage(info, montage):
         The measurement info to update.
     montage : instance of Montage
         The montage to apply.
+    ch_type : string
+        The type of electrode contained in this montage. Can be 'eeg' or 'ieeg'
     """
-    if not _contains_ch_type(info, 'eeg'):
+    supported_types = ('eeg', 'ieeg')
+    if ch_type not in supported_types:
+        raise ValueError('Montage type not supported')
```
Comment (agramfort) [inline mne/channels/layout.py]: ValueError('Channel type should be eeg or ieeg (got %s).' % ch_type)

File: mne/channels/layout.py [hunk 6]
```diff
@@ -927,6 +933,10 @@ def apply_montage(info, montage):
         info['chs'][ch_idx]['loc'] = np.r_[pos, [0.] * 9]
         sensors_found = True
 
+        if ch_type == 'ieeg':
+            info['chs'][ch_idx]['kind'] = FIFF.FIFFV_SEEG_CH
+            info['chs'][ch_idx]['coord_frame'] = FIFF.FIFFV_COORD_MRI
```
Comment (agramfort) [inline mne/channels/layout.py]: you're mixing objectives here. we have another function to change channel types. See rename_channels function.

File: mne/channels/tests/test_montage.py [hunk 7]
```diff
@@ -80,3 +81,36 @@ def test_montage():
     assert_array_equal(pos2, montage.pos)
     assert_array_equal(pos3, montage.pos)
     assert_equal(montage.ch_names, info['ch_names'])
+
+
+def test_ieeg_montage():
+    """Test an iEEG montage"""
```
Comment (agramfort) [inline mne/channels/tests/test_montage.py]: we use either seeg or ieeg in the code base. We have not avoid this.

**Comment by aestrivex** [comment]
rename_channels isn't a sufficient API for this purpose, because it requires the user to manually specify the names of all the channels to be changed. Instead there needs to be a way to automatically change the type (and preferably also the coord_frame attribute) of all the channels specified in the montage.

Would it be more consistent with the existing API to create a different function to do this similar to rename_channels?

**Comment by dengemann** [comment]
We could extend the rename_channels API to allow for type-wise renaming, or some pattern matching.

**Comment by agramfort** [comment]
have a look at:

https://github.com/mne-tools/mne-python/pull/1825

it can be done in 1 line with the fix

**Comment by aestrivex** [comment]
Thanks, it looks like it will work for my use case.

I'll close this PR.

**Pull request closed without merging**